### PR TITLE
feat(new-widget-builder-experience): Port handleDefaultFields function - [DD-536]

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Port handleDefaultFields function
- [x] Port "Read the docs" button
- [x] Minor layout improvements (see comments)

### CODE REVIEW

File: static/app/views/dashboardsV2/widgetBuilder/header.tsx [hunk 1]
```diff
@@ -69,6 +69,12 @@ function Header({
           >
             {t('Give Feedback')}
           </Button>
+          <Button
+            external
+            href="https://docs.sentry.io/product/dashboards/custom-dashboards/#widget-builder"
+          >
+            {t('Read the docs')}
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/header.tsx:76]: Add 'Read the docs' button
image

File: static/app/views/eventsV2/table/columnEditCollection.tsx [hunk 2]
```diff
@@ -549,7 +549,7 @@ class ColumnEditCollection extends React.Component<Props, State> {
 
 const RowContainer = styled('div')`
   display: grid;
-  grid-template-columns: ${space(3)} 1fr ${space(3)};
+  grid-template-columns: ${space(3)} 1fr 40px;
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/table/columnEditCollection.tsx:552]: the size was not correct here and button was overflowing 

image

File: static/app/views/dashboardsV2/widgetCard/index.tsx [hunk 3]
```diff
@@ -223,6 +223,7 @@ const StyledPanel = styled(Panel, {
   min-height: 96px;
   display: flex;
   flex-direction: column;
+  overflow: hidden;
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetCard/index.tsx:226]: image

happens when the widget is loading

File: static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx [hunk 4]
```diff
@@ -183,79 +191,56 @@ function YAxisSelector({
...
-`;
-
 const QueryFieldWrapper = styled('div')`
   display: flex;
   align-items: center;
   justify-content: space-between;
-  margin-bottom: ${space(1)};
+
+  :not(:last-child) {
+    margin-bottom: ${space(1)};
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx:236]: image

the last item does not need to have a margin-bottom

File: static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx [hunk 5]
```diff
@@ -624,6 +682,12 @@ const PageContentWithoutPadding = styled(PageContent)`
   padding: 0;
 `;
 
+const VisualizationWrapper = styled('div')`
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx:685]: image

to be consistent

File: static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx [hunk 6]
```diff
@@ -459,19 +520,16 @@ function WidgetBuilder({
...
-                      );
-                    }}
+                    {({measurements}) => (
+                      <YAxisSelector
+                        widgetType={widgetType}
+                        displayType={state.displayType}
+                        fields={explodedFields}
+                        fieldOptions={getAmendedFieldOptions(measurements)}
+                        onChange={handleChangeYAxisOrColumnField}
+                        errors={state.errors?.queries}
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx:530]: we should double-check this

File: static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx [hunk 7]
```diff
@@ -260,51 +317,71 @@ function WidgetBuilder({
...
       const newQuery = cloneDeep(query);
-
       newQuery.fields = fieldStrings;
-
-      if (!aggregateAliasFieldStrings.includes(orderbyAggregateAliasField)) {
-        newQuery.orderby = '';
-
+      if (
+        !aggregateAliasFieldStrings.includes(orderbyAggregateAliasField) &&
+        query.orderby !== ''
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx:337]: fix the following issue:

```
interval: 1d
orderby: falseundefined
partial: 1
query: 
referrer: api.dashboards.widget.bar-chart
statsPeriod: 24h
yAxis: count%28%29
yAxis: count_unique%28device.locale%29
```

**Review by narsaynorath — APPROVED** [review]
I left a comment about `overflow: hidden` in the widget card causing the acceptance tests to fail, but this is looks good to me

There are some merge conflicts from the previous PR, I didn't want to push into your PR so I made a separate branch where I resolved the conflicts. If the diff looks good to you then you can merge it into this branch without the headache. It's at `nar/feat/new-widget-builder-experience-port-y-axis-part-2`

[The diff is here](https://github.com/getsentry/sentry/compare/feat/new-widget-builder-experience-port-y-axis...nar/feat/new-widget-builder-experience-port-y-axis-part-2).

I'll delete that branch from the repo later so that name isn't cluttering things.

File: static/app/views/dashboardsV2/widgetCard/index.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/views/dashboardsV2/widgetCard/index.tsx:226]: oh noo :( 

good catch @narsaynorath thank you! sure I'll remove the overflow and we can maybe address this in a separate PR

**Comment by priscilawebdev** [comment]
thank you @narsaynorath 🌮

**Pull request merged**